### PR TITLE
mkosi: Update to latest

### PR DESCRIPTION
**Pull request by daandemeyer**
Configuration now takes priority over CLI options so we have to configure the defaults for settings that we want to allow overriding from the CLI. We also explicitly set some other settings so that they can't be overridden from the CLI anymore. For example the base and initrd image should never be made bootable so we set Bootable=no explicitly for both.

### CODE REVIEW

File: mkosi.conf.d/05-qemu-mem.conf [hunk 1]
```diff
@@ -8,4 +8,4 @@ Distribution=|arch
 Distribution=|centos
 
 [Host]
-QemuMem=3G
+@QemuMem=3G
```
Comment (bluca) [inline mkosi.conf.d/05-qemu-mem.conf:11]: what's `@`?

Comment (daandemeyer) [reply mkosi.conf.d/05-qemu-mem.conf:11]: Sets the default value which is used if no value is specified explicitly.

**Review by keszybz — APPROVED** [review]
I'm not convinced that this is the right approach. But it's not something to discuss here. Mkosi has been changed to work like this, so systemd must follow suit. Those changes seem correct.

**Comment by keszybz** [comment]
```
Run mkosi --debug genkey
‣ Including configuration file /home/runner/work/systemd/systemd/mkosi.conf
‣ Including configuration file /home/runner/work/systemd/systemd/mkosi.conf.d/05-qemu-mem.conf
‣ Including configuration file /home/runner/work/systemd/systemd/mkosi.conf.d/10-centos.conf
‣ Including configuration file /home/runner/work/systemd/systemd/mkosi.conf.d/10-systemd.conf
‣ build/ does not exist
Traceback (most recent call last):
  File "/home/runner/work/_actions/systemd/mkosi/b4a2a7da64707b9ca6bedb93e7026d4e46f3182e/mkosi/__main__.py", line 20, in propagate_failed_return
    yield
  File "/usr/lib/python3.10/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/home/runner/work/_actions/systemd/mkosi/b4a2a7da64707b9ca6bedb93e7026d4e46f3182e/mkosi/__main__.py", line 45, in main
    args, presets = parse_config(sys.argv[1:])
  File "/home/runner/work/_actions/systemd/mkosi/b4a2a7da64707b9ca6bedb93e7026d4e46f3182e/mkosi/config.py", line 2016, in parse_config
    parse_config(Path("."), namespace, defaults)
  File "/home/runner/work/_actions/systemd/mkosi/b4a2a7da64707b9ca6bedb93e7026d4e46f3182e/mkosi/config.py", line 1969, in parse_config
    parse_config(p if p.is_file() else Path("."), namespace, defaults)
  File "/home/runner/work/_actions/systemd/mkosi/b4a2a7da64707b9ca6bedb93e7026d4e46f3182e/mkosi/config.py", line 1946, in parse_config
    setattr(ns, s.dest, s.parse(v, getattr(ns, s.dest, None)))
  File "/home/runner/work/_actions/systemd/mkosi/b4a2a7da64707b9ca6bedb93e7026d4e46f3182e/mkosi/config.py", line 384, in config_parse_list
    return new + [parse(v) for v in values if v]
  File "/home/runner/work/_actions/systemd/mkosi/b4a2a7da64707b9ca6bedb93e7026d4e46f3182e/mkosi/config.py", line 384, in <listcomp>
    return new + [parse(v) for v in values if v]
  File "/home/runner/work/_actions/systemd/mkosi/b4a2a7da64707b9ca6bedb93e7026d4e46f3182e/mkosi/config.py", line 167, in parse_path
    die(f"{value} does not exist")
  File "/home/runner/work/_actions/systemd/mkosi/b4a2a7da64707b9ca6bedb93e7026d4e46f3182e/mkosi/log.py", line 30, in die
    sys.exit(1)
SystemExit: 1
Error: Process completed with exit code 1.
```

**Review by bluca — APPROVED** [review]

**Comment by bluca** [comment]
```
 core downloading...
 extra downloading...
error: failed retrieving file 'extra.db' from geo.mirror.pkgbuild.com : The requested URL returned error: 404
error: failed to synchronize all databases (failed to retrieve some files)
```

**Pull request merged**